code/modules/mob/living/carbon/carbon_movement.dm

```diff
@@ -64,11 +64,22 @@
 	. = ..()
 	if(isnull(.))
 		return
-	if(. & !(FLYING | FLOATING))
+	if(!(. & (FLYING | FLOATING)))
```

Comment:
?

Reply:
What is the doubt?
The old check was not good, this should be checking if the movement type is not flying nor floating.

Reply:
ahhh I see.